src/enumerable.cr

```diff
@@ -705,7 +705,7 @@ module Enumerable(T)
   # [1, 2, 3].map { |i| i * 10 } # => [10, 20, 30]
   # ```
   def map(&block : T -> U) forall U
-    ary = [] of U
+    ary = Array(U).new(size)
```

Comment:
is not `size` is a function which have computation cost?

Reply:
No, `size` has only in very very rare cases a computation cost: when `Enumerables`'s `size` is used because it iterates through every element to find out the size. However on arrays for example, this will call `Array`'s `size` which is a `getter` and has zero computation cost.
https://github.com/crystal-lang/crystal/blob/60760a5460aa13a1f80c5a1a7410782dc4076b77/src/array.cr#L55